src/Grids/latitude_longitude_grid.jl

```diff
@@ -237,41 +240,46 @@ function with_precomputed_metrics(grid)
                                              Azᶠᶜᵃ, Azᶜᶠᵃ, Azᶠᶠᵃ, Azᶜᶜᵃ, grid.radius)
 end
 
-function validate_lat_lon_grid_args(FT, latitude, longitude, z, size, halo, topology, precompute_metrics)
-    Nλ, Nφ, Nz = N = size
-    
-    λ₁, λ₂ = get_domain_extent(longitude, Nλ)
-    @assert λ₁ <= λ₂ && λ₂ - λ₁ ≤ 360
-
-    φ₁, φ₂ = get_domain_extent(latitude, Nφ)
-    @assert -90 <= φ₁ <= φ₂ <= 90
-
-    (φ₁ == -90 || φ₂ == 90) &&
-        @warn "Are you sure you want to use a latitude-longitude grid with a grid point at the pole?"
-    
+function validate_lat_lon_grid_args(topology, size, halo, FT, latitude, longitude, z, precompute_metrics)
+        
     if !isnothing(topology)
-        TX, TY, TZ = topology
-        Nλ, Nφ, Nz = N = validate_size(TX, TY, TZ, size)
-        Hλ, Hφ, Hz = H = validate_halo(TX, TY, TZ, halo)
-    else
-        Lλ = λ₂ - λ₁
+        TX, TY, TZ = validate_topology(topology)
+        Nλ, Nφ, Nz = size = validate_size(TX, TY, TZ, size)
+    else # Set default topology according to longitude
+        Nλ, Nφ, Nz = size # using default topology, does not support Flat
+        λ₁, λ₂ = get_domain_extent(longitude, Nλ)
 
+        Lλ = λ₂ - λ₁
         TX = Lλ == 360 ? Periodic : Bounded
         TY = Bounded
         TZ = Bounded
     end
 
+    # Validate longitude and latitude
+    λ₁, λ₂ = get_domain_extent(longitude, Nλ)
+    λ₂ - λ₁ ≤ 360 || throw(ArgumentError("Longitudinal extent cannot be greater than 360 degrees."))
+    λ₁ <= λ₂      || throw(ArgumentError("Longitudes must increase west to east."))
+
+    φ₁, φ₂ = get_domain_extent(latitude, Nφ)
+    -90 <= φ₁ || throw(ArgumentError("The southernmost latitude cannot be less tha -90 degrees."))
+    φ₂ <= 90  || throw(ArgumentError("The northern latitude cannot be less tha -90 degrees."))
+    φ₁ <= φ₂  || throw(ArgumentError("Latitudes must increase south to north."))
+
+    φ₁ == -90 && @warn("LatitudeLongitudeGrid will contain a singularity at the south pole.")
+    φ₂ == +90 && @warn("LatitudeLongitudeGrid will contain a singularity at the north pole.")
+        
     if TX == Flat || TY == Flat 
         precompute_metrics = false
     end
 
-    Hλ, Hφ, Hz = H = validate_halo(TX, TY, TZ, halo)
+    longitude = validate_dimension_specification(TX, longitude, :longitude, Nλ, FT)
+    latitude  = validate_dimension_specification(TY, latitude,  :latitude,  Nφ, FT)
+    z         = validate_dimension_specification(TZ, z,         :z,         Nz, FT)
 
-    longitude = validate_dimension_specification(TX, longitude, :x, Nλ, FT)
-    latitude  = validate_dimension_specification(TY, latitude,  :y, Nφ, FT)
-    z         = validate_dimension_specification(TZ, z,         :z, Nz, FT)
+    halo = validate_halo(TX, TY, TZ, halo)
+    topology = (TX, TY, TZ)
 
-    return Nλ, Nφ, Nz, Hλ, Hφ, Hz, latitude, longitude, z, (TX, TY, TZ), precompute_metrics
+    return topology, size, halo, latitude, longitude, z, precompute_metrics
```

Comment:
```suggestion
    return TX, TY, TZ, size, halo, longitude, latitude, z, precompute_metrics
```
to follow
https://github.com/CliMA/Oceananigans.jl/blob/8c42e9b127e7d07220a65051e0909122c8175086/src/Grids/rectilinear_grid.jl#L297?

Reply:
I feel its easier to read with fewer return args, I will change `rectilinear_grid` accordingly